src/main/java/io/supertokens/session/accessToken/AccessToken.java

```diff
@@ -435,12 +445,21 @@ static AccessTokenInfo fromJSON(AppIdentifier appIdentifier, JsonObject payload,
                 checkRequiredPropsExist(payload, version);
                 JsonObject userData = new JsonObject();
 
+                String[] protectedPropNames = getProtectedProps(version);
+
                 for (Map.Entry<String, JsonElement> element : payload.entrySet()) {
                     if (Arrays.stream(protectedPropNames).noneMatch(element.getKey()::equals)) {
                         userData.add(element.getKey(), element.getValue());
                     }
                 }
 
+                TenantIdentifier tenantIdentifier;
+                if (version == VERSION.V3) {
+                    tenantIdentifier = new TenantIdentifier(appIdentifier.getConnectionUriDomain(), appIdentifier.getAppId(), null);
+                } else {
+                    tenantIdentifier = new TenantIdentifier(appIdentifier.getConnectionUriDomain(), appIdentifier.getAppId(), payload.get("tId").getAsString());
```

Comment:
add a comment explaining why we are doing like this

Reply:
Done